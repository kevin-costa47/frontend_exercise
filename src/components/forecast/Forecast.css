.forecast_grid {
  display: grid;
  width: 100%;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 1rem;
}
