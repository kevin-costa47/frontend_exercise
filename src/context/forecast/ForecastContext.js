import { createContext } from "react";

const forecastContext = createContext({
    forecasts: [],
    region: {},
    regions: [],
});

export default forecastContext;
