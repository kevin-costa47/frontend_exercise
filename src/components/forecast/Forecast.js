import React, { useContext } from "react";
import ForecastContext from "../../context/forecast/ForecastContext";
import "./Forecast.css";

const Forecast = () => {
  const forecastContext = useContext(ForecastContext);
  const { forecasts, region } = forecastContext;

  if (forecasts.length === 0) {
    return null;
  }
  const { idAreaAviso } = region;

  return forecasts.map((forecast, index) => {
    const { forecastDate, tMax, tMin, precipitaProb } = forecast;
    const idArea = `${idAreaAviso} ${forecastDate}`;
    const tempMax = `${tMax}ºC`;
    const tempMin = `${tMin}ºC`;

    return (
      <div className="card" key={index}>
        <div className="forecast_grid">
          <h3 className="item-center">{idArea}</h3>
          <h3 className="item-center">
            <b>Max. Temperature:</b>
            {tempMax}
          </h3>
          <h3 className="item-center">
            <b>Min. Temperature:</b>
            {tempMin}
          </h3>
          <h3 className="item-center">
            <b>Precipitation Probability (%):</b>
            {precipitaProb}
          </h3>
        </div>
      </div>
    );
  });
};

Forecast.propTypes = {};

export default Forecast;
