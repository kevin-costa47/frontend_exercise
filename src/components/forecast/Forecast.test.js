import React from 'react';
import { shallow, mount } from 'enzyme';
import Forecast from './Forecast';
import ForecastContext from  '../../context/forecast/ForecastContext';

const renderWithContext = (children, value) => {
    return (
        <ForecastContext.Provider value={value}>
            { children }
        </ForecastContext.Provider>
    );
};

describe('Forecast', () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    const forecasts = [{
        tMax: '25',
        tMin: '20',
        precipitaProb: '30',
        forecastDate: 'Date'
    }];


    const region = {
        idAreaAviso: '23',
    };

    const context = {
        forecasts,
        region,
    };

    it('should render the Forecast Component correctly', () => {
        let wrapped = mount(renderWithContext(<Forecast/>, context));

        expect(wrapped).toMatchSnapshot();
    });

    it('should render null if has not forecasts', () => {
        const newContext = {
            ...context,
            forecasts: []
        };
        let wrapped = shallow(renderWithContext(<Forecast/>, newContext));

        expect(wrapped).toEqual({});
    });
});
