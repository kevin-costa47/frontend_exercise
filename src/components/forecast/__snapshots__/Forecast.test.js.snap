// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast should render the Forecast Component correctly 1`] = `
<Forecast>
  <div
    className="data_row"
    key="0"
  >
    <div
      className="grid_forecast"
    >
      <h3
        className="all-center"
      >
        <b />
        23 Date
      </h3>
      <h3
        className="all-center"
      >
        <b>
          Max. Temperature:
        </b>
        25ºC
      </h3>
      <h3
        className="all-center"
      >
        <b>
          Min. Temperature:
        </b>
        20ºC
      </h3>
      <h3
        className="all-center"
      >
        <b>
          Precipitation Probability (%):
        </b>
        30
      </h3>
    </div>
  </div>
</Forecast>
`;
